### ISSUE TITLE
Monitor for Linux Systemd services

### ISSUE THREAD

**Issue by yurividal** [louislam/uptime-kuma#1677]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Monitor

### 🔖 Feature description

It would be nice to be able to monitor services running on a linux instance.

### ✔️ Solution

Check running services on the host machine by running ````systemctl -q is-active````

**Comment by joerich** [comment]
I like the idea! is there a way to check a windows box too?

**Comment by Computroniks** [comment]
This seems like a use for the push monitor type. You could write a basic bash script (or a script in any other language) that runs `systemctl -q is-active` or it's windows alternative. Then just send a request to the push monitor endpoint. If you put this on a cron script then it should work fine.

**Comment by yurividal** [comment]
> This seems like a use for the push monitor type. You could write a basic bash script (or a script in any other language) that runs `systemctl -q is-active` or it's windows alternative. Then just send a request to the push monitor endpoint. If you put this on a cron script then it should work fine.

I get that this is possible with some extra scripting. But, if I were to write a script, I would use telegram-send to send the push notification and not need uptime-kuma at all.
(Which I already have done, by the way).
This feature request is so that, the everyday user (non-programmer) can use the functionality and monitor services.

**Comment by fattredd** [comment]
I'm not sure how you're proposing this would work. For uptime-kuma to execute this directly it would need ssh (or similar) access the remote machines. I think the security and configuration implications of that kind of defeat the point of script-avoidance,

Is that what you're thinking, or am I misunderstanding? Perhaps prewriting some cronjob snippets in the documentation would solve the problem?

**Comment by yurividal** [comment]
I was initially thinking of monitoring services on the UptimeKuma host only, but, I guess a built-in, transparent,  ssh access where the user would only fill username/password or key keyfile would take it to a new level. (AFAIK, that's exactly how software like PRTG monitor Linux hosts)

**Comment by DomiiBunn** [comment]
> I was initially thinking of monitoring services on the UptimeKuma host only, but, I guess a built-in, transparent, ssh access where the user would only fill username/password or key keyfile would take it to a new level. (AFAIK, that's exactly how software like PRTG monitor Linux hosts)

Or can use an agent that posts to an api the state of the service as that is a little bit more secure...

**Comment by yurividal** [comment]
Agents are a great idea!

**Comment by DomiiBunn** [comment]
I have this currently. It's not described or anything but it works. It can send a ping and or see if a process is running. 

I do plan to change it into an executable in the future but it works. Just add the kuma push listner and place that as a kuma endpoint with everything before the ? in the url. 

https://git.honex.life/domiibunn/kuma-push

**Comment by Computroniks** [comment]
> Agents are a great idea!

I think that this would take Uptime Kuma away from its core scope. It is meant to use the same systems as a end user to establish uptime e.g HTTP and ping. I think that adding agents to the core project would be too far out of scope and would turn the project into a more general system monitoring solution. If there is a requirement for something like this then I think that the push monitor is the solution.

**Comment by DomiiBunn** [comment]
...Exatly, But something needs to push to it regardless?

**Comment by Computroniks** [comment]
> ...Exatly, But something needs to push to it regardless?

My point it not including agents in the core project and instead letting end users create their own for their specific tasks using the push monitor endpoint. This helps to prevent chaos with trying to support many different systems and keeps the project from getting cluttered

**Comment by DomiiBunn** [comment]
True true, On that note a question, Does the push endpoint support anything else but url args, It does not work well with url encoded characters

**Comment by Computroniks** [comment]
I don't think so as I think that it is just a GET request to the endpoint, perhaps this could be something to think about in the future, adding the ability to provide a bit more info to the push monitor

**Comment by fionnb** [comment]
> Agents are a great idea!

Agents are a bad idea. If you want monitoring with agents, there are dozens of solutions out there already. Go check out Zabbix, for example. It can do everything and shows you how hard it is to create GUI that can handle all the complexity.

If you want a no-fuzz, no-clutter, quick-setup uptime checker, get Kuma. If not, get something bigger. Don't fuck Kuma up by stacking shit on top until it becomes just another clunky system monitor.

**Comment by Justinzobel** [comment]
I started doing this with Push type Monitors but realized it would take a lot of manual effort to keep them synchronized with what was running on my server, then multiply that by 100 servers.
